src/Switch-mbTechnologies_HVM/main.py

```diff
@@ -192,63 +210,75 @@ def get_identification(self):
         return answer
     
     def get_operation_complete(self):
-        """
-        Waits until all pending operations are complete then returns 1
-        
+        """Waits until all pending operations are complete then returns 1
+
+        Returns:
+            bool: True if operation complete finished. Otherwise it will timeout.
         """
         self.port.write("*OPC?")
-        if self.port.read() == "1":
-            return True
-        else:
-            return False
+        return self.port.read() == "1"
 
     def reset(self) -> None:
-        """
-        Resets the instrument and opens all relays. 
+        """Resets the instrument and opens all relays.
         """
         self.port.write("*RST")
+        if not self.port_string.startswith("GPIB"):
+            self.port.read()
 
-    def run_self_test(self) -> None:
-        """
-        Starts execution of the relay matrix self-test. Depending on the matrix size the test may take up to 1 minute to 
-        finish. 
-        Disconnect all measurement cables before starting the self-test! The test may fail if cables are connected. 
+    def start_self_test(self) -> None:
+        """Starts execution of the relay matrix self-test. Depending on the matrix size the test may take up to
+        1 minute to finish. Disconnect all measurement cables before starting the self-test! The test may fail if cables
+         are connected.
         """
         self.port.write("*TST")
+        if not self.port_string.startswith("GPIB"):
+            self.port.read()
 
     def enquiry_self_test(self) -> None:
-        """
-        Queries the result of the relay matrix self-test.
+        """Queries the result of the relay matrix self-test.
+
         Response: 
         0 Self-test passed 
         n (relay, …) Self-test failed , n=number of failing relays, list of failing relays follows 
         """
         self.port.write("*TST?")
+        return self.port.read()
+
+    def run_self_test(self) -> str:
+        self.start_self_test()
```

Comment:
The start_self_test() function has some important remarks in the docstring. If a user would use the run_self_test() function instead, he would not notice this information. Maybe it should also be mentioned here. And maybe a few more words on when a user should use start_self_test() vs run_self_test(). From the names of the functions they could mean the same.

Reply:
I added further docu and renamed enquiry_self_test